### ISSUE TITLE
Application fails to start if can't connect to collection host

### ISSUE THREAD

**Issue by bgroupe** [newrelic/elixir_agent#181]
**Describe the bug**
Similar to/Regression of #131
When starting an application during an APM outage at the time of this writing 2/25/2020 application fails to start presumably if host ping times out. Full exception:

~~~
00:16:11.478 [info] Application new_relic_agent exited: NewRelic.Application.start(:normal, []) returned an error: shutdown: failed to start child: NewRelic.EnabledSupervisor
    ** (EXIT) exited in: GenServer.call(NewRelic.Harvest.Collector.AgentRun, :ping, 5000)
        ** (EXIT) time out
[os_mon] memory supervisor port (memsup): Erlang has closed
[os_mon] cpu supervisor port (cpu_sup): Erlang has closed
{"Kernel pid terminated",application_controller,"{application_start_failure,new_relic_agent,{{shutdown,{failed_to_start_child,'Elixir.NewRelic.EnabledSupervisor',{'EXIT',{timeout,{'Elixir.GenServer',call,['Elixir.NewRelic.Harvest.Collector.AgentRun',ping,5000]}}}}},{'Elixir.NewRelic.Application',start,[normal,[]]}}}"}
Kernel pid terminated (application_controller) ({application_start_failure,new_relic_agent,{{shutdown,{failed_to_start_child,'Elixir.NewRelic.EnabledSupervisor',{'EXIT',{timeout,{'Elixir.GenServer',ca
Crash dump is being written to: erl_crash.dump...done
~~~

This is problematic for running applications with self-healing orchestrators ie Kubernetes. 

**Environment**
* Elixir Version: 1.10.0-otp-22
* Erlang Version: 22.2.4
* Agent version: 1.16.4

**Comment by binaryseed** [comment]
Thanks for this report! I'll take a closer look

**Comment by damonvjanis** [comment]
I'm seeing this same issue, but in local development and I can't actually use any version above 1.14.0 or I get that error on startup.

Here are my replication steps:

1. `mix new my_app`
2. Add `{:new_relic_agent, "~> 1.0"} to `mix.exs`
3. Add `config :new_relic_agent, app_name: "my_app", license_key: "<valid_license_key>"` to `config.exs`
4. Run `iex -S mix`, and it hangs starting up and then crashes with the error.

I've had a co-worker run the same on his machine and there's no crash (consistently), and on mine it crashes every time, so this is potentially difficult to replicate.

I'm using Elixir 1.8 and Erlang 21.2, using a `.tool-versions` file with asdf and my co-worker used the same versions so we're thinking it may not be related to Elixir / Erlang versions. I also replicated on Elixir 1.9 on my machine, and I can't find any setup that lets me actually run an app with the package at 1.15 or above on my computer.

**Comment by damonvjanis** [comment]
Ok we've traced this down a bit for why this is happening in local development for me. There's a function `NewRelic.Util.maybe_add_fqdn/1` that is making this call: `:net_adm.dns_hostname(:net_adm.localhost())`. On my Macbook Pro (2017, 15", 16 RAM) it takes 5 seconds for this call to return, and because it takes so long and it's being run from within a `handle_continue` clause in the genserver, the ping doesn't return in the 5000 millisecond timeout period and the error the OP is showing gets raised.

Still investigating why that particular call is so slow on my machine and isn't on other machines.

**Comment by binaryseed** [comment]
When this issue was first created there was an incident occurring that impacted the server the agent connects to. I intend to re-work that ping check so that it does not impact the whole app (which was always the goal)

**Comment by binaryseed** [comment]
Fixed with https://github.com/newrelic/elixir_agent/pull/187 and released as `1.16.6`

Thanks for the report!
